packages/components/src/internal/form.ts

```diff
@@ -290,10 +306,35 @@ export class FormControlController implements ReactiveController {
     }
   }
 
+  /** Checks for the presence of the attributes 'data-user-valid' or 'data-user-invalid' on the host form element and updates its corresponding style state. */
+  updateValidityStyle() {
+    if (this.host.hasAttribute('data-user-valid') && this.host.checkValidity()) {
+      // check for presence of showValidStyle attribute before mutating
+      if (this.host.showValidStyle !== undefined) this.host.showValidStyle = true;
+      this.host.showInvalidStyle = false;
+    } else if (this.host.hasAttribute('data-user-invalid') && !this.host.checkValidity()) {
+      if (this.host.showValidStyle !== undefined) this.host.showValidStyle = false;
+      this.host.showInvalidStyle = true;
+    } else {
+      if (this.host.showValidStyle !== undefined) this.host.showValidStyle = false;
+      this.host.showInvalidStyle = false;
+    }
+  }
+
   /** Returns the associated `<form>` element, if one exists. */
   getForm() {
     return this.form ?? null;
   }
+  /** Returns a styled `<div>` element to display inline errors. */
+  renderErrorMessage() {
```

Comment:
As far as I can tell, there is no downside to rendering the error message like this. All designs match and the attributes are always the same.  Instead of providing a class, I took this approach.  Thoughts?